packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -281,7 +290,16 @@ class WidgetInspectorService {
         selection.current = object;
       }
       if (selectionChangedCallback != null) {
-        selectionChangedCallback();
+        if (WidgetsBinding.instance.schedulerPhase == SchedulerPhase.idle) {
+          selectionChangedCallback();
+        } else {
+          // It isn't safe to trigger the selection change callback if we are in
+          // the middle of rendering the frame.
+          WidgetsBinding.instance.scheduleTask(
```

Comment:
probably better to use SchedulerBinding here (the result is the same but it's more idiomatic)

Reply:
done